cmd/lora/main.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	r "github.com/go-redis/redis"
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/lora"
+	"github.com/mainflux/mainflux/lora/api"
+	pahoBroker "github.com/mainflux/mainflux/lora/paho"
+
+	kitprometheus "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/mainflux/mainflux/lora/redis"
+	"github.com/nats-io/go-nats"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	defHTTPPort     = "8180"
+	defLoraMsgURL   = "tcp://localhost:1883"
+	defNatsURL      = nats.DefaultURL
+	defLogLevel     = "error"
+	defESURL        = "localhost:6379"
+	defESPass       = ""
+	defESDB         = "0"
+	defInstanceName = "lora"
+	defRouteMapURL  = "localhost:6379"
+	defRouteMapPass = ""
+	defRouteMapDB   = "0"
+
+	envHTTPPort     = "MF_LORA_ADAPTER_HTTP_PORT"
+	envLoraMsgURL   = "MF_LORA_ADAPTER_LORA_MESSAGE_URL"
+	envLoraServURL  = "MF_LORA_ADAPTER_LORA_SERVER_URL"
+	envNatsURL      = "MF_NATS_URL"
+	envLogLevel     = "MF_LORA_ADAPTER_LOG_LEVEL"
+	envESURL        = "MF_THINGS_ES_URL"
+	envESPass       = "MF_THINGS_ES_PASS"
+	envESDB         = "MF_THINGS_ES_DB"
+	envInstanceName = "MF_LORA_ADAPTER_INSTANCE_NAME"
+	envRouteMapURL  = "MF_LORA_ADAPTER_ROUTEMAP_URL"
+	envRouteMapPass = "MF_LORA_ADAPTER_ROUTEMAP_PASS"
+	envRouteMapDB   = "MF_LORA_ADAPTER_ROUTEMAP_DB"
+)
+
+type config struct {
+	httpPort     string
+	loraMsgURL   string
```

Comment:
You're missing `loraServURL` here. Add it in `config`, or remove it from env vars.